staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -2407,7 +2415,7 @@ func RunTestGuaranteedUpdateWithSuggestionAndConflict(ctx context.Context, t *te
 	err := store.GuaranteedUpdate(ctx, key, updatedPod, false, nil,
 		storage.SimpleUpdate(func(obj runtime.Object) (runtime.Object, error) {
 			pod := obj.(*example.Pod)
-			pod.Name = "foo-2"
+			pod.Generation = 2
```

Comment:
This tests were somewhat broken for two reasons:
- changing name is unrealistic (forbidden) and watchcache is lost - switched to use Generation instead
- suggestion can be ignored by implementation - relaxed the validation check below

Reply:
was `RunTestGuaranteedUpdateWithSuggestionAndConflict` flaky or did you change it to match reality ?

Reply:
since we are changing the test to match reality would it make sense to add a new annotation instead of messing with the Generation field which is set by the system ?

Reply:
actually, the Generation will be updated during update before we call this function so it is okay to mess with that field.

Reply:
generation isn't handled automatically by the system - we're handing it manually at strategy level for individual resources. So no - I think this is good.